contracts/transfer/src/state.rs

```diff
@@ -201,6 +201,60 @@ impl TransferState {
         self.mint_withdrawal("WITHDRAW", withdraw);
     }
 
+    /// Takes the deposit addressed to this contract, and immediately withdraws
+    /// it, effectively performing an atomic conversion between Phoenix notes
+    /// and Moonlight balance.
+    ///
+    /// This functions checks whether the deposit included with the transaction
+    /// is the exact value included in `convert`, and imposes that the
+    /// caller is indeed this contract.
+    ///
+    /// # Panics
+    /// This can only be called by this contract - the transfer contract - and
+    /// will panic if this is not the case.
+    pub fn convert(&mut self, convert: Withdraw) {
+        if rusk_abi::caller() != TRANSFER_CONTRACT {
+            panic!("Only a direct call can be a conversion");
+        }
+        if *convert.contract() != TRANSFER_CONTRACT.to_bytes() {
+            panic!("The conversion must target the transfer contract");
+        }
+
+        let deposit = transitory::deposit_info_mut();
+        match deposit {
+            Deposit::Available(_, deposit_value) => {
+                let deposit_value = *deposit_value;
+
+                if convert.value() != deposit_value {
+                    panic!("The value to convert doesn't match the value in the transaction");
+                }
+
+                // This is a direct contract call, as the first check in this
```

Comment:
Does the `direct contract call` mean the same as in the comment below `first contract call`. If so, I'd use the same nomenclature in both places. If not, does it mean contract self-call? Pls. explain :)

Reply:
It does mean the same. The result of not having well laid out terminology. Will use just one